core/src/commonMain/kotlin/com/powersync/PowerSyncDatabase.kt

```diff
@@ -25,6 +26,11 @@ interface PowerSyncDatabase : ReadQueries, WriteQueries {
      */
     val currentStatus: SyncStatus
 
+    /**
+     * @returns a StateFlow which can be used to process sync state updates
+     */
+    fun getStatusFlow(): StateFlow<SyncStatus>
```

Comment:
A common pattern is to have the class, `SyncStatus` in this case, have a method `asFlow` to provide a flow

This is an example from[SQLDelight](https://github.com/cashapp/sqldelight/blob/master/extensions/coroutines-extensions/src/commonMain/kotlin/app/cash/sqldelight/coroutines/FlowExtensions.kt#L37) adding `asFlow` to the `Query` class

Reply:
I updated the implementation to now have the `SyncStatus` provide a flow which emits data updates each time the status has been changed.